### DISCUSSION TITLE
How is the card populated in createPaymentMethod()

### DISCUSSION THREAD

**Discussion by lbk9** [flutter-stripe/flutter_stripe#110] [General]
From the example project it is not clear how the card details are passed into the createPaymentMethod function. I am attempting to migrate from the stripe_payment package that is now discontinued. In the old package it was clear and simple on how to attach credit card details to the call. 

Can someone let me know how I do this with this plugin? It seems that it somehow attaches the details from the UI. I already have a UI created when integrating with the previous plugin and I'd rather not have to change it. Is there a way I can manually populate the card details that get sent off to create a payment method?

Thanks

**Comment by jonasbark** [comment]
Hi Liam,

using custom card details is not currently possible as discussed in https://github.com/flutter-stripe/flutter_stripe/issues/40  due to PCI compliance issues. We're searching for a feasible solution here.

The way it works currently is that you enter the card details in the CardField and the information always stays on the Android / iOS side and never leaves in the direction of Flutter or vice versa. That means, in order to create the payment method, you'll need to enter the complete card information to the CardField. The examples will help you determine that.

**Reply by lbk9** [reply]
Thanks Jonas. I would switch over to using this widget however the current verison has a fatal crash on iOS that has been raised here #87

**Reply by lbk9** [reply]
Further to this, does that mean that the stripe_payment widget is not PCI compliant?

**Reply by jonasbark** [reply]
I would assume it isn't.

**Reply by jamesblasco** [reply]
Any code that processes personal card information and has not been audited by a 3rd party is not PCI compliant by default. And that means every year you have to make sure to do all the paperwork to demonstrate your company is PCI compliant.

The easiest way is to use what Stripe already gives you PCI compliant. That is the case of the CardField or the PaymentSheet.

I have submitted a PR to fix #87, so that should not be a problem soon
